clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -4245,20 +4244,17 @@ def CIR_AtomicCmpXchg : CIR_Op<"atomic.cmpxchg", [
                        UnitAttr:$is_volatile);
 
   let assemblyFormat = [{
-    `(`
-      $ptr `:` qualified(type($ptr)) `,`
-      $expected `:` type($expected) `,`
-      $desired `:` type($desired) `,`
-      `success` `=`  $succ_order `,`
-      `failure` `=`  $fail_order
-    `)`
-    (`align` `(` $alignment^ `)`)?
     (`weak` $weak^)?
+    custom<AtomicCmpXchgMemOrder>($succ_order, $fail_order)
```

Comment:
does this need custom printer/parser?
why not just something like:

success ( $succ_order) failure($fail_order)

Reply:
The intention here is to avoid showing the same memory order twice when `$succ_order` and `$fail_order` are the same.